### DISCUSSION TITLE
Which OS and version of the is best for the luftdatenpumpe

### DISCUSSION THREAD

**Discussion by ohobby** [earthobservations/luftdatenpumpe#29] [Q&A]
Hello,
which OS in which version is best to use for the luftdatenpumpe.
What other software must be provided before so that the installation works without problems?

Thanks,
Oliver

**Comment by amotl** [comment]
Hi Oliver,

thanks for writing in.

> Which OS in which version is best to use for the Luftdatenpumpe.

A POSIX OS will probably be the best choice to run Luftdatenpumpe, we are running it on Linux and the development happens on macOS.

> What other software must be provided before so that the installation works without problems?

Luftdatenpumpe is pretty standalone. However, it also has adapters to store timeseries data into InfluxDB and station information into PostGIS. On top of that, there are some example resources (data source and dashboard JSON files) you can import into Grafana for visualizing stuff. There, for displaying maps, you will probably find [Grafana Map Panel](https://community.panodata.org/t/grafana-map-panel/121) useful.

You are invited to have a look at [1,2], where some of these things are running in production.

With kind regards,
Andreas.

[1] https://weather.hiveeyes.org/grafana/d/PyFBQZiik/foderierte-wetter-und-luftdaten
[2] https://weather.hiveeyes.org/grafana/dashboards/f/NFeP70fik/luft-ldi

**Reply by ohobby** [reply]
Hi Andreas,
it was already clear to me that it should run on a Linux, which distribution and version is favored so that it runs cleanly with Python and the databases?
I wanted to see if it is easy to load only the historical data of Frankfurt, or if it is easier to load all historical data and then filter it later.
Kind regards,
Oliver

**Reply by amotl** [reply]
Hi Oliver,

while we are fans of Debian Linux, Luftdatenpumpe should work on any version of Linux with a sane Python 3. I just verified the installation on my macOS machine with Python 3.8 and Python 3.9. We strongly recommend to install it within a [virtualenv](https://docs.python.org/3/library/venv.html) in order not to pollute the Python installation shipped with your distribution.

Like:
```sh
# Setup
python -m venv .venv
source .venv/bin/activate
pip install luftdatenpumpe

# Testdrive
luftdatenpumpe --version
luftdatenpumpe 0.20.2
```

Just let us know if you encounter any problems installing it.

With kind regards,
Andreas.

**Reply by ohobby** [reply]
Hi Andreas,

i make a clean Debian Buster install.
At the luftdatenpumpe install i have some errors:

```
Building wheels for collected packages: luftdatenpumpe, SQLAlchemy-Utils, docopt, dogpile.cache, geohash2, munch, paho-mqtt, py-openaq, rfc3339, alembic, odfpy, Mako, et-xmlfile
  Running setup.py bdist_wheel for luftdatenpumpe ... error
  
error: invalid command 'bdist_wheel'
```

`luftdatenpumpe --version` output are:
```
/root/.venv/lib/python3.7/site-packages/sqlalchemy/sql/functions.py:67: SAWarning: The GenericFunction 'array_agg' is already registered and is going to be overridden.
  "is going to be overridden.".format(identifier)
luftdatenpumpe 0.20.2
```

- How to fix the error: invalid command 'bdist_wheel'?
- How to fix the SAWarning?

Thanks,
Oliver

**Reply by amotl** [reply]
Hi Oliver,

I tried re-running the installation on a Debian "buster" Docker image like

```sh
docker run -it --rm amd64/debian:buster-slim bash
apt-get update
apt-get install python3 python3-pip --yes

$ python3 -V
Python 3.7.3

$ pip3 --version
pip 18.1 from /usr/lib/python3/dist-packages/pip (python 3.7)

pip3 install luftdatenpumpe
```

and haven't been able to observe the `error: invalid command 'bdist_wheel'`. Maybe the version of `pip` is outdated or your installation is lacking the `wheel` package, see [1]. However, I can confirm the warning coming from SQLAlchemy, see #30.

Nevertheless, both things should not do any harm when trying to operate Luftdatenpumpe as the installation apparently was successful. Please let us know if this is not the case when you encounter any problems during operation.

With kind regards,
Andreas.

[1] https://stackoverflow.com/questions/34819221/why-is-python-setup-py-saying-invalid-command-bdist-wheel-on-travis-ci

**Comment by amotl** [comment]
Hi Oliver,

thank you for all the valuable feedback on the documentation of Luftdatenpumpe while you recently installed it on your systems. We truly appreciate it.

At [^1], we finally published the rendered version of the documentation, including a few improvements on the spots you mentioned. If you can spot any bugs, or have any other suggestions for improvements, we will be happy to hear about them.

With kind regards,
Andreas.

[^1]: https://luftdatenpumpe.readthedocs.io/
